portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -79,22 +95,41 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(contentType);
     String name = getParameter(request, "name", "Anonymous");
-    String comment = getParameter(request, "comment", "");
+    String comment = getParameter(request, "text", "");
     Date createDate = new Date();
 
     if (comment == "" || comment == null) {
       throw new IOException("Comment is blank");
     }
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("name", name);
-    commentEntity.setProperty("text", comment);
-    commentEntity.setProperty("date", createDate);
-
-    datastore.put(commentEntity);
-
-    response.sendRedirect("/");
+    String idTokenString = request.getParameter("id_token");
```

Comment:
Please explicitly handle the case when `id_token` parameter is missing. It'd be much better than letting the code throw an NPE and then catching it in a generic try-catch block.

Reply:
See also: https://github.com/twitter-archive/commons/blob/master/src/java/com/twitter/common/styleguide.md#exceptions

Reply:
Tried to make error handling more narrow and explicit.